src/shared/dissect-image.c

```diff
@@ -2272,7 +2272,9 @@ int dissected_image_acquire_metadata(DissectedImage *m) {
         /* As per the os-release spec, if the image is an extension it will have a file
          * named after the image name in extension-release.d/ */
         if (m->image_name) {
-                char *ext = strjoina("/usr/lib/extension-release.d/extension-release.", m->image_name, "0");
+                char *ext;
+
+                ext = strjoina("/usr/lib/extension-release.d/extension-release.", m->image_name, "0");
```

Comment:
Thanks for fixing this - one day I will stop getting alloca&friends wrong. One day :-)

Reply:
But does it really matter? I undestand why doing alloca inside of a function call is not allowed. But here, we don't care if `ext` is allocated first, or the contents first. Either way, all the compiler has to do, is allocate two objects on the stack. Could it get that wrong?

Reply:
Does this matter because strjoina() is a Statement Expression macro?

It's unclear to say the least...

Reply:
there's a certain amount of weirdness in all of this: alloca() allocations are supposed to survive until end of the block, but normal stack variables are not...